File: src/main/java/no/nav/veilarboppfolging/controller/response/OppfolgingEnhetMedVeilederResponse.java
package no.nav.veilarboppfolging.controller.response;

import lombok.Data;
import lombok.Value;
import lombok.experimental.Accessors;

@Data
@Accessors(chain = true)
public class OppfolgingEnhetMedVeilederResponse {
    Oppfolgingsenhet oppfolgingsenhet;
    String veilederId;
    String formidlingsgruppe;
    String servicegruppe;
    String hovedmaalkode;

    @Value
    public static class Oppfolgingsenhet {
        String navn;
        String enhetId;
    }
}

File: src/main/kotlin/no/nav/veilarboppfolging/oppfolgingsbruker/arena/OppfolgingEnhetMedVeilederResponse.kt
package no.nav.veilarboppfolging.oppfolgingsbruker.arena


sealed class GetOppfolginsstatusResult
class GetOppfolginsstatusSuccess(
    val result: OppfolgingEnhetMedVeilederResponse
): GetOppfolginsstatusResult()
class GetOppfolginsstatusFailure(
    val error: Exception
): GetOppfolginsstatusResult()


data class OppfolgingEnhetMedVeilederResponse(
     val oppfolgingsenhet: Oppfolgingsenhet,
     val veilederId: String,
     val formidlingsgruppe: String,
     val servicegruppe: String,
     val hovedmaalkode: String,
) {
    data class Oppfolgingsenhet(
        val navn: String,
        val enhetId: String,
    )
}


